### PR TITLE
Fix issue with scalars and __rpow__

### LINKED ISSUE #16685: Incorrect size for __rpow__(scalar, float)
## 🐛 Bug

```py
>>> 4 ** torch.tensor(0.5)
torch.tensor([2.])
```

## To Reproduce

Here is a regression test:
```py
def test_rpow_scalar_float_shape():
    assert (4 ** torch.tensor(0.5)).shape == ()
```

## Expected behavior

Binary op of (scalar, number) should be a scalar.

## Environment

```
PyTorch version: 1.0.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.14.1
GCC version: Could not collect
CMake version: version 3.12.0

Python version: 2.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] Could not collect
[conda] mkl                       2018.0.2                      1
[conda] mkl_fft                   1.0.1            py27h917ab60_0
[conda] mkl_random                1.0.1            py27h78cc56f_0
[conda] torch                     1.0.0                     <pip>
[conda] torch-nightly             1.0.0.dev20181206           <pip>
[conda] torchfile                 0.1.0                     <pip>
[conda] torchvision               0.2.1                     <pip>
```

### PR DESCRIPTION
**Pull request by vishwakftw**
Changelog:

- Modify __rpow__ function in tensor.py to adapt to scalars

Test plan:

- Add scalar entry in test_rpow
- Add CUDA port

Fixes #16685

### CODE REVIEW

**Comment by vishwakftw** [comment]
cc: @soumith @fritzo

File: torch/tensor.py [hunk 1]
```diff
@@ -388,7 +388,7 @@ def __ipow__(self, other):
         raise NotImplementedError("in-place pow not implemented")
 
     def __rpow__(self, other):
-        return self.new([other]) ** self
+        return self.new_tensor(other) ** self
```
Comment (SsnL) [inline torch/tensor.py]: shouldn't this be `as_tensor`? Also, `new_tensor` throws a warning when the input is already a tensor...

Comment (vishwakftw) [reply torch/tensor.py]: If `other` is a tensor, then this would be dispatch to the __pow__ function instead right?

Comment (SsnL) [reply torch/tensor.py]: Oh you are correct! Sorry!

Comment (vishwakftw) [reply torch/tensor.py]: No worries :smile:

**Pull request closed without merging**